blocks/bricks/base.py

```diff
@@ -738,13 +838,53 @@ def init(self, *args, **kwargs):
 
 class VariableRole(object):
     """A collection of constants referring to variable roles."""
-    COST = "cost"
-    INPUT = "input"
-    OUTPUT = "output"
-    MONITOR = "monitor"
+    #: Any variable attached to a brick or application call
+    AUXILIARY = 'auxiliary'
+    #: A scalar variable which represents some cost or regularization penalty
+    COST = 'cost'
+    #: The input to a brick
+    INPUT = 'input'
+    #: The output of a brick
+    OUTPUT = 'output'
+    #: Any parameter of the model
+    PARAMETER = 'parameter'
+    #: The weights of a particular linear transformation
+    WEIGHTS = 'weights'
+    #: The biases added after a linear transformation
+    BIASES = 'biases'
+
+    @classmethod
+    def add_role(cls, var, role):
```

Comment:
In fact this could be achieved by a hierarchy of roles:

```
class Parameter(VariableRole):
     pass

class Weights(Parameter):
     pass
```